src/onboarding/steps.ts

```diff
@@ -0,0 +1,214 @@
+    }
+    currentScreen = s
+  }
+
+  const navigateHome = () => {
+    currentScreen = Screens.WalletHome
+  }
+
+  // @ts-ignore: Compiler doesn't understand that navigate() can update currentScreen
+  while (currentScreen !== END_OF_ONBOARDING_SCREEN) {
+    const stepInfo = _getStepInfo({
+      firstScreenInStep: currentScreen,
+      navigator: {
+        navigate,
+        popToScreen: () => {
+          // no-op
+        },
+        navigateHome,
+        navigateClearingStack: navigate,
+      },
+      dispatch: () => {
+        // no-op
+      },
+      props: onboardingProps,
+    })
+    if (!stepInfo) {
+      throw new Error(`No step info found for ${currentScreen}.`)
+    }
+    stepInfo?.next()
+  }
+
+  return {
+    totalSteps: totalCounter,
+    step: stepCounter,
+  }
+}
+
+export function goToNextOnboardingScreen({
+  firstScreenInCurrentStep,
+  onboardingProps,
+}: {
+  firstScreenInCurrentStep: Screens
+  onboardingProps: OnboardingProps
+}) {
+  const stepInfo = _getStepInfo({
+    firstScreenInStep: firstScreenInCurrentStep,
+    navigator: {
+      navigate: NavigationService.navigate,
+      popToScreen: NavigationService.popToScreen,
+      navigateHome: NavigationService.navigateHome,
+      navigateClearingStack: NavigationService.navigateClearingStack,
+    },
+    dispatch: store.dispatch,
+    props: onboardingProps,
+  })
+  stepInfo?.next()
+}
+
+export function _getStepInfo({ firstScreenInStep, navigator, dispatch, props }: GetStepInfoProps) {
+  const { navigate, popToScreen, navigateHome, navigateClearingStack } = navigator
+  const {
+    recoveringFromStoreWipe,
+    choseToRestoreAccount,
+    supportedBiometryType,
+    skipVerification,
+    numberAlreadyVerifiedCentrally,
+  } = props
+
+  switch (firstScreenInStep) {
+    case Screens.NameAndPicture:
+      return {
+        next: () => {
+          if (recoveringFromStoreWipe) {
+            navigate(Screens.ImportWallet)
+          } else {
+            navigate(Screens.PincodeSet)
+          }
+        },
+      }
+    case Screens.PincodeSet:
+      return {
+        next: () => {
+          if (supportedBiometryType !== null) {
+            navigate(Screens.EnableBiometry)
+          } else if (choseToRestoreAccount) {
+            popToScreen(Screens.Welcome)
+            navigate(Screens.ImportWallet)
+          } else if (skipVerification) {
+            dispatch(initializeAccount())
+            // Tell the app that the user has already seen verification so that it
+            // doesn't prompt for verification after the app is killed. This same function
+            // is called when the user manually skips verification on the verification screen.
+            dispatch(setHasSeenVerificationNux(true))
+            navigateHome()
+          } else {
+            navigateClearingStack(Screens.VerificationStartScreen)
+          }
+        },
+      }
+    case Screens.EnableBiometry:
+      return {
+        next: () => {
```

Comment:
Why is the return signature of this function an object with a single property? Can we just return the `next` function directly?